### PR TITLE
Fix out of bounds bug

### LINKED ISSUE #5: Last entry with 0% go out of bounds
I have a perf file that finishes:
```
     0.00%     0.00%  grep             [kernel.kallsyms]           [k] fill_pmd
     0.00%     0.00%  pharo            [kernel.kallsyms]           [k] nmi_cpu_backtrace_handler
     0.00%     0.00%  pharo            [kernel.kallsyms]           [k] nmi_handle
     0.00%     0.00%  sh               [kernel.kallsyms]           [k] ghes_notify_nmi
     0.00%     0.00%  sh               [kernel.kallsyms]           [k] nmi_cpu_backtrace
     0.00%     0.00%  grep             [kernel.kallsyms]           [k] native_sched_clock
     0.00%     0.00%  perf_5.10        [kernel.kallsyms]           [k] native_sched_clock
     0.00%     0.00%  pharo            [kernel.kallsyms]           [k] intel_bts_enable_local
     0.00%     0.00%  sh               [kernel.kallsyms]           [k] native_sched_clock
     0.00%     0.00%  sh               [kernel.kallsyms]           [k] intel_bts_enable_local
     0.00%     0.00%  grep             [kernel.kallsyms]           [k] intel_bts_enable_local


#
# (Tip: If you prefer Intel style assembly, try: perf annotate -M intel)
#
```

The problem seems to be in `PerfTreeParser >> findChildrenIndexOfIndex: aNodeIndex` when `aNodeIndex = last index`

it does:

```st
....
	(self percentageOf: aNodeIndex) = (self percentageOf: aNodeIndex + 1)
		ifTrue: [
			everyChildren add: aNodeIndex + 1.
			^ everyChildren ].
....
```

And the percentage of a none existing line is always 0 😢 
```st
percentageOf: aLine

	| stringLine numbers |
	(aLine between: 1 and: self numberOfFunctions) ifFalse: [ ^ 0 ].
....
```

### PR DESCRIPTION
**Pull request by matburnx**
Fix #5, now if the `NodeIndex + 1` is out of bounds, the parent has no child 
Fix another issue I found where one line trees had the function below as child even though they are not related
```
     0.00%     0.00%  sh               [kernel.kallsyms]           [k] ghes_notify_nmi
     0.00%     0.00%  sh               [kernel.kallsyms]           [k] nmi_cpu_backtrace
```
For example, in this case `nmi_cpu_backtrace` would have been considered as the child of `ghes_notify_nmi` but they clearly are not (at least in this case) related

### CODE REVIEW

**Comment by guillep** [comment]
Ohoh! We need a test! :)

**Comment by matburnx** [comment]
I've added one and changed another one due to the changes in the test file being done on another PR

File: src/PerfTreeParser/PerfTreeParser.class.st [hunk 1]
```diff
@@ -70,10 +70,14 @@ PerfTreeParser >> findChildrenIndexOfIndex: aNodeIndex [
...
-			everyChildren add: aNodeIndex + 1.
-			^ everyChildren ].
+	(aNodeIndex + 1 between: 1 and: self numberOfFunctions) ifFalse: [
+		^ everyChildren ].
+
+	((self percentageOf: aNodeIndex)
+	 = (self percentageOf: aNodeIndex + 1) and: [
+		 ((self readLine: aNodeIndex + 1) findString: '[') = 0 ]) ifTrue: [
+		everyChildren add: aNodeIndex + 1.
+		^ everyChildren ].
```
Comment (guillep) [inline src/PerfTreeParser/PerfTreeParser.class.st:80]: We should sit and pair a bit on this code ^^

**Review by guillep — APPROVED** [review]
Thanks @matburnx !

**Pull request merged**